### PR TITLE
Issue84 - Formatting options for properties and accessors

### PR DESCRIPTION
**Pull request by Pilchie**
Add formatting options for braces in properties

### CODE REVIEW

**Comment by Pilchie** [comment]
Tagging @dotnet/mlangide @mkrueger @basoundr @heejaechang.

**Comment by heejaechang** [comment]
:+1:

File: src/VisualStudio/CSharp/Impl/Options/AutomationObject.cs [hunk 1]
```diff
@@ -194,6 +194,18 @@ public int NewLines_Braces_Method
             set { SetBooleanOption(CSharpFormattingOptions.NewLinesForBracesInMethods, value); }
         }
 
+        public int NewLines_Braces_Property
+        {
+            get { return GetBooleanOption(CSharpFormattingOptions.NewLinesForBracesInProperties); }
+            set { SetBooleanOption(CSharpFormattingOptions.NewLinesForBracesInProperties, value); }
+        }
+
+        public int NewLines_Braces_Accessor
```
Comment (DustinCampbell) [inline src/VisualStudio/CSharp/Impl/Options/AutomationObject.cs]: event accessors too?

File: src/VisualStudio/CSharp/Impl/CSharpVSResources.resx [hunk 2]
```diff
@@ -402,4 +402,10 @@
   <data name="Option_RenameTrackingPreview" xml:space="preserve">
     <value>Show preview for _rename tracking</value>
   </data>
+  <data name="NewLinesForBracesAccessors" xml:space="preserve">
+    <value>Place open brace on new line for property accessors</value>
```
Comment (DustinCampbell) [inline src/VisualStudio/CSharp/Impl/CSharpVSResources.resx]: Should this also say indexer and events?

File: src/VisualStudio/CSharp/Impl/Options/AutomationObject.cs (hunk 1 above)
Comment (Pilchie) [inline src/VisualStudio/CSharp/Impl/Options/AutomationObject.cs]: And Indexers?  Do you have a good suggestion for the name?

**Comment by DustinCampbell** [comment]
I think we should update the text of the option to mention indexers and events, but barring that: :+1:

File: src/VisualStudio/CSharp/Impl/CSharpVSResources.resx (hunk 2 above)
Comment (Pilchie) [inline src/VisualStudio/CSharp/Impl/CSharpVSResources.resx]: Updated to "Place open brace on new line for property, indexer, and event accessors"

File: src/VisualStudio/CSharp/Impl/Options/AutomationObject.cs (hunk 1 above)
Comment (mkrueger) [inline src/VisualStudio/CSharp/Impl/Options/AutomationObject.cs]: ATM I would suggest using just the Property braces option. It's very easy to let the options explode when you're going the fine grained route. At the end 80% of the options are not used :/.

Comment (Pilchie) [inline src/VisualStudio/CSharp/Impl/Options/AutomationObject.cs]: Yeah, I waffled a bit.  As the one who added most of the options VS has back in VS2005, I know about them not being used much, but this seemed like such an obvious next step I just did it.

**Comment by fschneidereit** [comment]
:+1: A big thumbs up to you guys for (finally) implementing this. I don't know how many hours I've wasted manually reformatting properties and events in K&R style in the past. Thanks a lot!

**Pull request merged**